base/intfuncs.jl

```diff
@@ -490,14 +490,15 @@ julia> prevpow(4, 16)
 16
 ```
 """
-function prevpow(a::Real, x::Real)
+function prevpow(a::Real, x::T) where T <: Real
     x < 1 && throw(DomainError(x, "`x` must be ≥ 1."))
     # See comment in nextpos() for a == special case.
     a == 2 && isa(x, Integer) && return _prevpow2(x)
     a <= 1 && throw(DomainError(a, "`a` must be greater than 1."))
     n = floor(Integer,log(a, x))
-    p = a^(n+1)
-    p <= x ? p : a^n
+    p = a^n
+    wp = widemul(p,a)
```

Comment:
This doesn't really need the widemul, only the overflow bit (which is free). Does LLVM do the appropriate transformation?

Reply:
I could fix this better, but given that I removed an integer pow, this will remain faster. If you have a good way to get the overflow bit, feel free to edit the PR. Otherwise I'll probably merge as is.

Reply:
```julia
julia> Base.mul_with_overflow(9,10)
(90, false)

julia> Base.mul_with_overflow(9,typemax(Int))
(9223372036854775799, true)
```

Reply:
This can't use `mul_with_overflow` since it needs to work with non-integers.

Reply:
Just branch on the type of `a`?